core/src/main/java/org/infinispan/util/AbstractDelegatingCacheStream.java

```diff
@@ -318,6 +324,11 @@ public void close() {
    }
 
    @Override
+   public <R1, A> R1 collect(SerializableSupplier<Collector<? super R, A, R1>> supplier) {
+      return collect(new CollectorSupplier2<>(supplier));
```

Comment:
Why not just call `return collect(CacheCollectors.serializableCollector(supplier));` ? Then we won't need the additional inner class.

Reply:
Doesn't work because the types don't match up:

    Error:(328, 37) java: method serializableCollector in class org.infinispan.stream.CacheCollectors cannot be applied to given types;
    required: org.infinispan.util.function.SerializableSupplier<java.util.stream.Collector<T,?,R>>
    found: org.infinispan.util.function.SerializableSupplier<java.util.stream.Collector<? super R,A,R1>>
    reason: cannot infer type-variable(s) T,R
    (argument mismatch;
    org.infinispan.util.function.SerializableSupplier<java.util.stream.Collector<? super R,A,R1>> 
    cannot be converted to
    org.infinispan.util.function.SerializableSupplier<java.util.stream.Collector<T,?,R>>)

To avoid causing much havoc to start with, I created `CollectorSupplier2` that's fully typed. CollectorSupplier is not fully typed on typed returned by the Supplier. I do agree both should use same collector supplier eventually... I'll play with it.